src/topic_training/src/Begin_Here.md

```diff
@@ -0,0 +1,122 @@
+
+Publishers and Subscribers do not need to be in separate files. In fact, if the Publisher and Subscriber rely on each other's data, it may be better to keep them in the same file. An example of the Publisher and Subscriber residing in the same file will be provided in the next section.
+
+## Setting up a Publisher and Subscriber
+Within this directory, you will find two files called Demo(.cpp or .py). These files will show a basic class containing both a Publisher and Subscriber. There will be comments throughout the files that explain the purpose of specific sections of the code.
+
+Note that there are quite a few sections of code that are marked as "Optional." You do not need to include everything in the examples when creating your own code. They are simply showing one way of setting up a Publisher and Subscriber in the same file.
+
+One file will be written in C++, and the other will be in Python. I recommend discussing with one of the Software Leads to determine which part of the rover you would like to be assigned to. This will help you decide which language to focus on for this training. However, being familiar with both languages will not hurt.
+
+Go through one of the files, making sure you understand all the code and comments. Come back to this file once you feel ready to move on.
+
+## Running a Publisher and Subscriber
+Before you are able to run the example code, find the CMakeLists.txt file that is outside of the src directory but still inside of the topic_training directory. 
+
+This file is mostly commented out. Make sure these are uncommented:
+
+- find_package(catkin REQUIRED
+    roscpp
+    rospy
+    std_msgs
+    message_generation
+  )
+
+- catkin_package(
+    CATKIN_DEPENDS message_runtime
+  )
+
+- include_directories(
+    ${catkin_INCLUDE_DIRS}
+  )
+
+The C++ and the Python file both require these to work.
+
+If you chose C++, also uncomment these:
+
+- add_executable(Demo src/Demo.cpp)
+  add_executable(PlotDemo src/PlotDemo.cpp)
+  add_executable(MotorChallenge src/MotorChallenge.cpp)
+
+- add_dependencies(Demo 
+    ${${PROJECT_NAME}_EXPORTED_TARGETS} 
+    ${catkin_EXPORTED_TARGETS}
+  )
+  add_dependencies(PlotDemo 
+    ${${PROJECT_NAME}_EXPORTED_TARGETS} 
+    ${catkin_EXPORTED_TARGETS}
+  )
+  add_dependencies(MotorChallenge 
+    ${${PROJECT_NAME}_EXPORTED_TARGETS} 
+    ${catkin_EXPORTED_TARGETS}
+  )
+
+- target_link_libraries(Demo 
+    ${catkin_LIBRARIES}
+  )
+  target_link_libraries(PlotDemo
+    ${catkin_LIBRARIES}
+  )
+  target_link_libraries(MotorChallenge
+    ${catkin_LIBRARIES}
+  )
+
+If you chose Python, also uncomment this:
+
+- catkin_install_python(PROGRAMS src/Demo.py src/PlotDemo.py src/MotorChallenge.py
+    DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
+  )
```

Comment:
You could uncomment all of these.  CMake is complicated at a first pass, so I would avoid making large, unexplained changes like these.  Maybe a reference or explanation?  Otherwise, avoid this kind of change altogether.

Reply:
Fixed in 08a9c40.